src/averageAges.js

```diff
@@ -15,11 +16,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people.filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century)
+    : people.filter((p) => p.sex === 'm');
+
+  let sum = 0;
+
+  men.forEach((m) => sum += m.died - m.born);
+
+  return sum/men.length;
```

Comment:
1. Use destructuring to access object properties; 
2. Use the reduce method and const variable declaration to improve readability and performance;
```suggestion
  const men = century
    ? people.filter(({ sex, died }) => 
       sex === 'm' && Math.ceil(died / 100) === century
      )
    : people.filter(({ sex }) => sex === 'm');

  const sum = men.reduce((acc, { died, born }) => {
     acc += died - born;
     
     return acc;
  }, 0);

  return sum / men.length;
```